src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -121,9 +169,11 @@ const useFetchChannelHook = () => {
   const saveAllChannelData = async (channels, channelCategory: ChannelCategory) => {
     const filteredChannels = filterChannels(channels);
 
-    filteredChannels?.forEach(async (channel) => {
-      await saveChannelData(channel, channelCategory);
+    const channelPromises = filteredChannels?.map((channel) => {
+      return Promise.resolve(saveChannelData(channel, channelCategory));
     });
+
+    await Promise.all(channelPromises);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made here are significant and improve the code quality. The use of `Promise.all` to wait for all channel data to be saved is a good practice as it can help prevent blocking of the main thread. However, it's important to ensure that error handling is properly implemented in case any of the promises reject.


<!-- This is an auto-generated comment by OSS CodeRabbit -->